### PR TITLE
HSEARCH-4924 Follow-up: minor improvements to incremental builds

### PR DESCRIPTION
**Pull request by yrodiere**
https://hibernate.atlassian.net/browse/HSEARCH-4924

Some minor syntax changes + we'll mark stages as skipped if the incremental builds skips all relevant modules.

I only tested the various options/commands locally, I'm hoping to test it for real in another PR.

### CODE REVIEW

File: Jenkinsfile [hunk 1]
```diff
@@ -921,6 +929,11 @@ void mavenNonDefaultBuild(BuildEnvironment buildEnv, String args, List<String> a
 					--fail-at-end \
 					$argsWithProjectSelection \
 	"""
+
+	if (incrementalBuild && 0 == sh(script: "test ! -s '${incrementalProjectsListFile}'", returnStatus: true)) {
```
Comment (marko-bekhta) [inline Jenkinsfile:933]: maybe we could do that before running the tests ^ . Or we' want the output of this exact build to determine if the stage needs to be marked as skipped?

Comment (yrodiere) [reply Jenkinsfile:933]: Well the file is created by gib during the Maven build, and we're only interested in it if it's empty, i.e. if the Maven build did close to nothing... so I don't think it's worth a separate Maven execution?

Comment (marko-bekhta) [reply Jenkinsfile:935]: thanks 😃 
yeah it seemed confusing, running the build to mark it skipped, but yeah it makes sense now 😃, we basically want  to see that nothing was tested on CI side without looking into the logs and so on, right ?

Comment (yrodiere) [reply Jenkinsfile:937]: Right:

```
	// In incremental builds, the Maven execution above
	// created a file listing projects relevant to the incremental build.
	// If it is empty, it means the incremental build didn't actually do anything,
	// so make sure to mark the stage as skipped.
```

**Pull request merged**